ext/storage_sources/local_store/local_store.c

```diff
@@ -1238,6 +1239,7 @@ wiredtiger_extension_init(WT_CONNECTION *connection, WT_CONFIG_ARG *config)
       (ret = pthread_rwlock_init(&local->flush_lock, NULL)) != 0) {
         (void)local_err(local, NULL, ret, "pthread_rwlock_init");
         free(local);
+        return (ret);
```

Comment:
Needs to return out of here. Otherwise, it'll fall through and mess with the freed memory.